### ISSUE TITLE
name conflict with redux?

### ISSUE THREAD

**Issue by mattkrick** [rofrischmann/react-fela#7]
I like how this wrapper mimics the redux api, but I'm concerned that also using a `connect` HOC is going to force folks to `import {connect as fela} from 'fela'` in every single doc as a best practice. Any thoughts on using something with a less generic name to make import statements a little cleaner?

**Comment by rofrischmann** [comment]
Heyhey,
Actually I did this to enforce the presentational/container pattern. If you strictly use that together with Redux, you would never have Fela & Redux within the same component as Fela styles your presentational components and Redux handles state within your container components. 

I am not sure if I want to rename that, but it seems fair enough to discuss here. Also because it might confuse ppl using Redux and Fela together for the first time.

**Comment by mattkrick** [comment]
TL;DR: I should be able to know what the import refers to by just the name, not by where it comes from. 

If I have to look up at the `import` statement to verify that it's coming from fela instead of redux, I might as well rename it in every file so i have something a little more expressive.  

i absolutely agree you should never see redux & fela imported in the same file. However, I'd liken it to calling a bunch of your components `Component`. Just because they're coming from different places and you'll never see them imported into the same file (`/headerButton/Component`, `/footerButton/Component`, `/sidebarLink/Component`) doesn't mean they should all have the same name. Ideally, react-redux would have been a little more expressive with its naming (`connect` is pretty vague), but that ship has already sailed.

**Comment by cpeddecord** [comment]
If needed, I imagine you could do something akin to this:

```
import { compose } from 'redux';
import { connect as felaConnect } from 'react-fela';
import { connect as reduxConnect } from 'react-redux';

function MyComponent() { ... }

export default compose(
  felaConnect(mapStylesToProps),
  reduxConnect(mapStateToProps)
)(MyComponent);
```

I don't mind the `connect` naming as much as `<Provider />`, both in this library and react-redux. React-redux's `<Provider />` component is ill-named whereas I feel React Intl got it right by naming their provider `<IntlProvider />`. Horses for courses, but I'd be happy to make the changes needed.

**Comment by rogchap** [comment]
Just want to add my 2¢ that having the same naming as redux is confusing: When I was looking through the example code to see the implementation (I read code before the docs) I thought that Fela was using the redux to store the styles (in the redux store)... and I thought wow... that's different :) 
Obviously, i just looked at the `connect` and `Provider` without looking at the import statements.
`connect` for presentation vs container is possibly ok, but you are more likely to have both Provider for redux and fela on your root component:
```javascript
import { Provider } from 'react-redux'
import { Provider as StyleProvider } from 'react-fela'
// ...
<Provider store={store}>
  <StyleProvider renderer={renderer}>
    <Router routes={routes} />
  </StyleProvider>
</Provider>
```

If my 2¢ is worth anything, I vote for a rename (at least for `Provider`).

**Comment by rofrischmann** [comment]
Heyo guys!

I see that ppl confuse Felas bindings. I am indeed not using the Redux store hehe. I thought about renaming both. How about `RendererProvider` and just `fela` instead of `connect`?

**Comment by rofrischmann** [comment]
This will most likely be the next breaking change for a 4.0 version. I will add it to the main repo as I close all other issue boards for fela related packages.
